test/mocks.ts

```diff
@@ -84,3 +84,16 @@ export const getServicesMock = () => {
     }
   }
 }
+
+export const getListenerMock = () => {
+  const listener = {
```

Comment:
you can do this instead (i think)

listenerMock = mock(listener.prototype)

means it will always use the correct API. But it doesn't change anything otherwise